M/MKL/build_tarballs.jl

```diff
@@ -27,10 +27,10 @@ fi
 """
 
 platforms = [
-    Linux(:x86_64, libc=:glibc),
-    MacOS(:x86_64),
-    Windows(:i686),
-    Windows(:x86_64),
+    Platform("x86_64", "linux"; libc="glibc"),
+    Platform("x86_64", "macos"),
+    Platform("i686", "windows"),
+    Platform("x86_64", "windows"),
 ]
```

Comment:
I don't think these platforms are actually used, can probably be removed

Reply:
Sure, but if we start on that way, there are tons more that likely should be changed? E.g. I see a bunch of platform lists where it really looks to me that they should have just taken all platforms and e.g. removed Windows. This should all indeed be cleaned up, but this PR only contains changes made by a script. As such, the changes are completely mechanical and unambiguous. While these cleanups now have a chance of introducing mistakes etc., so I'd prefer to try to do them in a separate PR (or multiple).

Reply:
I mean that this variable `platforms` is unused in this file.

Reply:
sure I got that, but where do you draw the limit in terms of how many "other people's mistakes" I need to fix before this can go in?

In this particular case, I actually think the better fix would be to keep `platforms` (or indeed perhaps change it to use `supported_platforms` and then change `no_autofix_platforms` and `autofix_platforms` below to something like this:
```
no_autofix_platforms = filter(!Sys.islinux, supported_platforms())
autofix_platforms = filter(Sys.islinux, supported_platforms())
```
Of course I can also remove `platforms`, but it really feels like an editorial decision to me, which I'd rather not make. If you insist, I will of course remove it, but...

Reply:
Ok, let's keep it this way here